app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :author, class_name: 'User', foreign_key: 'user_id'
+
+  def increment_user_post_counter
+    user.increment!(:post_counter)
+  end
```

Comment:
[OPTIONAL] It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) -* `after_save` to call the `#update_post_counter`  (in your case `#increment_user_post_counter`) method every time a new post has been saved to the database, then make the `update_post_counter` method private.

For example:
```
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```